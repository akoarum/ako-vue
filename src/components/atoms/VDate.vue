<template>
  <time class="vDate" :datetime="datetime">{{ viewDate }}</time>
</template>

<script lang="ts">
import Vue from 'vue'

export default Vue.extend({
  props: {
    datetime: { type: Date, required: true }
  },
  computed: {
    viewDate(): string {
      let dateString: string = ''
      const date = (this.datetime as unknown) as Date

      dateString += `${date.getFullYear()}年`
      dateString += `${date.getMonth() + 1}月`
      dateString += `${date.getDate()}日 `
      dateString += `${date.getHours()}時`
      dateString += `${date.getMinutes()}分`

      return dateString
    }
  }
})
</script>

<style lang="scss" scoped>
.vDate {
  font-size: 1.4rem;
  line-height: 1.2;
  color: #484848;
}
</style>
