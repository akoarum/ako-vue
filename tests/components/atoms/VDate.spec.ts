import { shallowMount } from '@vue/test-utils'
import VDate from '~/components/atoms/VDate.vue'

describe('VDate', () => {
  let wrapper: any

  beforeEach(() => {
    wrapper = shallowMount(VDate, {
      propsData: {}
    })
  })

  describe('initialized', () => {
    it('mountable', () => {
      expect(wrapper.isVueInstance()).toBe(true)
    })

    it('snapshot', () => {
      expect(wrapper.element).toMatchSnapshot()
    })
  })
})
