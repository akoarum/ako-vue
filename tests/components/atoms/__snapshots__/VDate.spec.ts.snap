// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VDate initialized snapshot 1`] = `
<time
  class="vDate"
  datetime="Tue Jan 01 2019 09:00:00 GMT+0900 (GMT+09:00)"
>
  2019年1月1日 9時0分
</time>
`;
